File: src/main/java/social/alone/server/user/UserDto.java
package social.alone.server.user;


import social.alone.server.interest.InterestDto;
import lombok.*;

import javax.validation.constraints.Email;
import javax.validation.constraints.Size;
import java.util.List;

@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(exclude = {"interests"})
public class UserDto {

  @Size(max = 20)
  private String name;

  @Email
  private String email;

  @Size(max = 3)
  private List<InterestDto> interests;

}

File: src/main/kotlin/social/alone/server/user/dto/UserDto.kt
package social.alone.server.user.dto


import social.alone.server.interest.InterestDto
import javax.validation.constraints.Email
import javax.validation.constraints.Size

data class UserDto(
        @field:Size(max = 20) val name: String? = null,
        @field:Email val email: String? = null,
        @field:Size(max = 3) val interests: List<InterestDto>? = null
)
